manager-bundle/src/HttpKernel/ContaoKernel.php

```diff
@@ -252,15 +252,20 @@ public static function fromRequest(string $projectDir, Request $request): HttpKe
     {
         self::loadEnv($projectDir, 'jwt');
 
-        // See https://github.com/symfony/recipes/blob/master/symfony/framework-bundle/4.2/public/index.php
-        if ($trustedProxies = $_SERVER['TRUSTED_PROXIES'] ?? null) {
-            Request::setTrustedProxies(explode(',', $trustedProxies), Request::HEADER_X_FORWARDED_ALL ^ Request::HEADER_X_FORWARDED_HOST);
-        }
-
         if ($trustedHosts = $_SERVER['TRUSTED_HOSTS'] ?? null) {
             Request::setTrustedHosts(explode(',', $trustedHosts));
         }
 
+        if ($trustedProxies = $_SERVER['TRUSTED_PROXIES'] ?? null) {
+            $trustedHeaderSet = Request::HEADER_X_FORWARDED_FOR | Request::HEADER_X_FORWARDED_PORT | Request::HEADER_X_FORWARDED_PROTO;
+
+            if ($trustedHosts) {
+                $trustedHeaderSet |= Request::HEADER_X_FORWARDED_HOST;
```

Comment:
Why are we doing this and [Symfony is not](https://github.com/symfony/recipes/blob/master/symfony/framework-bundle/4.2/public/index.php#L15-L21)?

Reply:
Because I think we can be smarter? :D Or in other words, tell me how am I supposed to configure my `.env` no to trust a certain host? You can't because it's never read from `X-Forwarded-Host`.
If you want to trust a certain host by setting `TRUSTED_HOSTS` in your `.env` I think you want to read it from there?